### PR TITLE
fix: fix uploader server read closed channel

### PR DESCRIPTION
**Pull request by sysvm**
### Description

Fix uploader server read closed channel.

### Rationale

In some extreme cases, uploader server may read data from closed channel.

### Example

N/A

### Changes

Notable changes: 
* N/A

### CODE REVIEW

File: base/gfspapp/upload_server.go [hunk 1]
```diff
@@ -129,10 +131,14 @@ func (g *GfSpBaseApp) GfSpUploadObject(stream gfspserver.GfSpUploadService_GfSpU
...
 		return nil
-	case <-initCh:
+	case <-initChan:
 		log.CtxDebug(ctx, "received first upload stream data")
-	case <-errChan:
-		return err
+	case err1, ok := <-errChan:
+		if !ok {
+			log.Error("error channel has been closed")
+			return fmt.Errorf("closed error channel")
```
Comment (guagualvcha) [inline base/gfspapp/upload_server.go:139]: It is very likely to return error here even the go routine end properly.

Comment (sysvm) [reply base/gfspapp/upload_server.go]: fixed

File: base/gfspapp/upload_server.go [hunk 2]
```diff
@@ -129,10 +127,11 @@ func (g *GfSpBaseApp) GfSpUploadObject(stream gfspserver.GfSpUploadService_GfSpU
 	select {
 	case <-ctx.Done():
 		return nil
-	case <-initCh:
+	case <-initChan:
 		log.CtxDebug(ctx, "received first upload stream data")
-	case <-errChan:
-		return err
+	case err1 := <-errChan:
+		log.Errorw("failed to upload object", "error", err)
```
Comment (will-2012) [inline base/gfspapp/upload_server.go:133]: err1?

File: base/gfspapp/upload_server.go [hunk 3]
```diff
@@ -246,8 +242,9 @@ func (g *GfSpBaseApp) GfSpResumableUploadObject(stream gfspserver.GfSpUploadServ
 		return nil
 	case <-initCh:
 		log.CtxDebug(ctx, "received first resumable upload stream data")
-	case <-errChan:
-		return err
+	case err1 := <-errChan:
+		log.Errorw("failed to resumable upload object", "error", err)
```
Comment (will-2012) [inline base/gfspapp/upload_server.go:246]: Ditto.

**Review by will-2012 — APPROVED** [review]

File: base/gfspapp/upload_server.go (hunk 2 above)
Comment (sysvm) [reply base/gfspapp/upload_server.go]: fixed

File: base/gfspapp/upload_server.go (hunk 3 above)
Comment (sysvm) [reply base/gfspapp/upload_server.go]: fixed

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**